jax/_src/ops/special.py

```diff
@@ -95,3 +97,22 @@ def logsumexp(a: ArrayLike, axis: Axis = None, b: ArrayLike | None = None,
     with jax.debug_nans(False):
       out = jnp.where(sign < 0, jnp.array(np.nan, dtype=out.dtype), out)
   return out
+
+@logsumexp.defjvp
+def _logsumexp_jvp(axis, keepdims, return_sign, where, primals, tangents):
+  a, b = primals
+  a_dot, b_dot = tangents
+  out = logsumexp(a, axis, b, keepdims, return_sign, where)
+
+  if return_sign:
+    out, sign = out
+
+  if b is None:
+    b = jnp.ones_like(a)
```

Comment:
We probably need to handle the case where `b_dot` is `None` as well?

Reply:
Good point. Am I right that `b_dot` is only None if `b` is None?

Reply:
Done.